code/modules/projectiles/guns/energy/experimental.dm

```diff
@@ -48,3 +50,122 @@
 	else
 		icon_state = "bison0"
 	return
+
+#define SPUR_FULL_POWER 4
+#define SPUR_HIGH_POWER 3
+#define SPUR_MEDIUM_POWER 2
+#define SPUR_LOW_POWER 1
+#define SPUR_NO_POWER 0
+
+/obj/item/weapon/gun/energy/spur
+	name = "\improper Spur"
+	desc = "A masterpiece crafted by the legendary gunsmith of a far-away planet."
+	icon = 'icons/obj/gun_experimental.dmi'
+	icon_state = "spur"
+	item_state = null
+	origin_tech = null
+	projectile_type = "/obj/item/projectile/spur"
+	charge_cost = 100
+	cell_type = "/obj/item/weapon/cell"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns_experimental.dmi', "right_hand" = 'icons/mob/in-hand/right/guns_experimental.dmi')
+	fire_delay = 0
+	recoil = 1
+	var/charge_tick = 0
+	var/firelevel = SPUR_FULL_POWER
+
+/obj/item/weapon/gun/energy/spur/New()
+	..()
+	processing_objects.Add(src)
+
+
+/obj/item/weapon/gun/energy/spur/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+/obj/item/weapon/gun/energy/spur/process()
+	charge_tick++
+	if(charge_tick < 2) return 0
+	charge_tick = 0
+	if(!power_supply) return 0
+	power_supply.give(100)
+	levelChange()
+	return 1
+
+/obj/item/weapon/gun/energy/spur/afterattack(atom/A as mob|obj|turf|area, mob/living/user as mob|obj, flag, params, struggle = 0)
+	levelChange()
+	..()
+
+/obj/item/weapon/gun/energy/spur/proc/levelChange()
+	var/maxlevel = power_supply.maxcharge
+	var/level = power_supply.charge
+	var/newlevel = 0
+	if(level == maxlevel)
```

Comment:
the if()s of a switch have to be constant expressions. maxlevel isn't a constant expression. I don't see how I can replace that if chain with a switch.